avm/res/network/network-manager/security-admin-configuration/main.bicep

```diff
@@ -59,3 +55,70 @@ output resourceId string = securityAdminConfigurations.id
 
 @sys.description('The resource group the security admin configuration was deployed into.')
 output resourceGroupName string = resourceGroup().name
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type applyOnNetworkIntentPolicyBasedServicesType = ('None' | 'All' | 'AllowRulesOnly')[]
+
+type ruleCollectionType = {
+  @sys.description('Required. The name of the admin rule collection.')
+  name: string
+
+  @sys.description('Optional. A description of the admin rule collection.')
+  description: string?
+
+  @sys.description('Required. List of network groups for configuration. An admin rule collection must be associated to at least one network group.')
+  appliesToGroups: {
+    @sys.description('Required. The resource ID of the network group.')
+    networkGroupId: string
```

Comment:
Should ideally be `networkGroupResourceId` though I wonder what a network group is to begin with. A `network security group`? If so, I recommend to call it such.

Please note, if changed to e.g. `networkSecurityGroupResourceId`, also the location where it is used must be updated wth a map function to map it to `the expected `networkGroupId`

Reply:
Also mind you, as you implemented the UDT in mutliple files, it would also need to be fixed in multiple files.

Looking forward to the time when types can simply be imported via a types.bicep ([ref](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-import))

Reply:
I would like to propose keeping it as, unless it's a hard requirement.

Reply:
Hey @ahmadabdalla, it would be as per the [spec](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr22---category-inputs---parametersvariables-for-resource-ids) and align with the other modules. But if you want, I can ask the team how 'strict'  we want to be about the rule, and if we can ignore it in cases.

Reply:
Hey @ahmadabdalla, is there a reason to want to keep it as `networkGroupId` instead of aligning to the spec?

Reply:
@AlexanderSehr @jtracey93 this has now been addressed and tested